src/main.js

```diff
@@ -821,10 +842,21 @@ class MiniGraphCard extends LitElement {
     const history = await this.getCache(entity.entity_id, this.config.useCompress);
     if (history && history.hours_to_show === this.config.hours_to_show) {
       stateHistory = history.data;
-      stateHistory = stateHistory.filter(item => new Date(item.last_changed) > initStart);
-      if (stateHistory.length > 0) {
+
+      let currDataIndex = stateHistory.findIndex(item => new Date(item.last_changed) > initStart);
+      if (currDataIndex !== -1) {
+        if (currDataIndex > 0) {
+          // include previous item
+          currDataIndex -= 1;
+          // but change it's last changed time
+          stateHistory[currDataIndex].last_changed = initStart;
+        }
```

Comment:
That is the actual bugfix described in the PR

Reply:
Very clever!
That should solve https://github.com/kalkih/mini-graph-card/issues/117 as well I believe?

Reply:
yes it looks like it is the same root cause